scipy/optimize/_qap.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+from . import linear_sum_assignment, OptimizeResult
+
+
+def umeyama_eigendecomposition(A, B, maximize=False):
+    """Finds an approximate solution to QAP using Umeyama's method."""
+    # 'Correct' non-hermitian matrices
+    if (A != A.T).any() or (B != B.T).any():
+        A = (A + A.T) / 2 + 1j * (A - A.T) / 2
+        B = (B + B.T) / 2 + 1j * (B - B.T) / 2
+
+    va = np.linalg.eigh(A)[1]
```

Comment:
SciPy returns eigenvalues (and corresponding eigenvectors) in increasing order, opposite the convention in the paper. This seems to affect the output for asymmetric matrices.

Reply:
Nice suggestion. I have swapped the eigenvector ordering and renamed the variables.